app/src/main/java/com/rockspin/androiddevtest/MyApplication.java

```diff
@@ -5,12 +5,23 @@
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 
+import java.util.List;
+
 import retrofit2.Retrofit;
 import retrofit2.converter.gson.GsonConverterFactory;
 
 public class MyApplication extends Application {
 
     private static APIService sAPIService;
+    private static MyApplication s_instance;
+    private List<CosmonautActivity> m_cosmonautActivities;
```

Comment:
because this activity points to a static list in the application class it is possible for the list to be changes outside of this class and for the adapter to have incorrect data.